### PR TITLE
test(dm): fix unstable test due to write relay log too slow

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/5287

### What is changed and how it works?

after syncer processes RowsEvent, check_sync_diff will pass, but XIDEvent may have not been written to relay log. We retry checking for 5 seconds.

#### Tests 

 - No code

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

**Comment by lance6716** [comment]
ping @buchuitoudegou @GMHDBJD

**Comment by lance6716** [comment]
ping @buchuitoudegou @GMHDBJD

**Comment by lance6716** [comment]
ping @buchuitoudegou @GMHDBJD

File: dm/tests/duplicate_event/run.sh [hunk 1]
```diff
@@ -73,11 +73,22 @@ function run_with_prepared_source_config() {
 	run_sql_source2 "show master status;"
 	binlog_file=$(grep "File" $TEST_DIR/sql_res.$TEST_NAME.txt | awk -F: '{print $2}' | xargs)
 	binlog_pos=$(grep "Position" $TEST_DIR/sql_res.$TEST_NAME.txt | awk -F: '{print $2}' | xargs)
-
 	server_uuid=$(tail -n 1 $WORK_DIR/worker2/relay-dir/server-uuid.index)
-	relay_log_size=$(ls -al $WORK_DIR/worker2/relay-dir/$server_uuid/$binlog_file | awk '{print $5}')
-	echo "binlog_pos: $binlog_pos relay_log_size: $relay_log_size"
-	[ "$binlog_pos" -eq "$relay_log_size" ]
+
+	succ=0
```
Comment (lonng) [inline dm/tests/duplicate_event/run.sh:78]: Seems that we just retry this case and make it run more times. Do we know the root cause of the relay log being too slow? Do we have another approach to speed up the relay log?

**Review by D3Hunter — APPROVED** [review]

File: dm/tests/duplicate_event/run.sh (hunk 1 above)
Comment (D3Hunter) [inline dm/tests/duplicate_event/run.sh:78]: wait the task `synced` in `query-status`?

Comment (lance6716) [reply dm/tests/duplicate_event/run.sh:78]: retry is expected. As in PR description, for the sequence [..., RowsEvent, XIDEvent], the test script will reach here after  relay writes RowsEvent and syncer processed RowsEvent. But relay may have no time to write XIDEvent.

Comment (lance6716) [reply dm/tests/duplicate_event/run.sh:78]: I just hope write XIDEvent will not exceed 5 seconds 😂

Comment (buchuitoudegou) [reply dm/tests/duplicate_event/run.sh:78]: waiting the syncer synced might be plausible.

**Review by buchuitoudegou — APPROVED** [review]

File: dm/tests/duplicate_event/run.sh (hunk 1 above)
Comment (lance6716) [reply dm/tests/duplicate_event/run.sh:78]: they will be also implemented as "check something and retry for a peroid". `synced` needs a DDL to trigger flush and needs to wait longer time.

**Comment by lance6716** [comment]
file change is included in another pr

**Pull request closed without merging**